server/handler.go

```diff
@@ -61,6 +61,10 @@ func (h *Handler) ConnectionClosed(c *mysql.Conn) {
 	delete(h.c, c.ConnectionID)
 	h.mu.Unlock()
 
+	if err := h.e.Catalog.UnlockTables(nil, c.ConnectionID); err != nil {
```

Comment:
In this case the tables would keep locked forever?

Reply:
If it fails, you mean?

Reply:
yes

Reply:
Then you can retry `UNLOCK TABLES`. If it fails it's because the tables `Unlock` failed, so nothing we can do there, as the implementation of that method depends on the tables.